bee-message/src/macro.rs

```diff
@@ -94,3 +94,29 @@ macro_rules! string_serde_impl {
         }
     };
 }
+
+/// A convenience macro to work around the fact the `[bitflags]` crate does not yet support iterating over the
+/// individual flags. This macro essentially creates the `[bitflags]` and puts the individual flags into a associated
+/// constant `pub const ALL_FLAGS: &'static []`.
+#[macro_export]
+macro_rules! create_bitflags {
+    ($(#[$meta:meta])* $Name:ident, $TagType:ty, [$(($FlagName:ident, $TypeName:ident),)+]) => {
+        bitflags! {
+            $(#[$meta])*
+            pub struct $Name: $TagType {
+                $(
+                    #[doc = "Signals the presence of a [`"]
+                    #[doc = stringify!($TypeName)]
+                    #[doc = "`]."]
```

Comment:
Look at `impl_id` you can concat the docs

Reply:
Ah, nice!